consensus/merge/src/main/java/org/hyperledger/besu/consensus/merge/MergeProtocolSchedule.java

```diff
@@ -93,7 +93,6 @@ private static ProtocolSpecBuilder applyMergeSpecificModifications(
             (gasCalculator, jdCacheConfig) ->
                 MainnetEVMs.paris(
                     gasCalculator, chainId.orElse(BigInteger.ZERO), EvmConfiguration.DEFAULT))
-        .blockProcessorBuilder(MergeBlockProcessor::new)
```

Comment:
Isn't the MergeBlockProcessor Optimized in terms of all the rewardCoinbase code path? If we remove this and start calling the rewardCoinbase method from the MainnetBlockProcessor aren't we executing more checks and code than we're supposed to?

Reply:
I don't think the MergeBlockProcessor is that optimised. We exit the rewardCoinbase method early due to https://github.com/hyperledger/besu/blob/main/ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/MainnetBlockProcessor.java#L55 with two very simple checks as opposed to the single isPostMerge check from the MergeBlockProcessor.

Reply:
If what you mentioned in the PR description is true: skipZeroBlockRewards=false then that if is probably skipped

Reply:
The skipZeroBlockRewards is set to true from spurious dragon onwards https://github.com/hyperledger/besu/blob/main/ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/MainnetProtocolSpecs.java#L267. So for Paris onwards where blockrewards=0 then we return immediately without trying apply rewards.

Reply:
You're are right, spurious dragon changes that to true, I was misled by the PR description.